From 29cc6511883e4b85dedd35504081fd01fbf1035b Mon Sep 17 00:00:00 2001
From: Ronny Lorenz <ronny@tbi.univie.ac.at>
Date: Tue, 8 Dec 2015 15:43:44 +0100
Subject: [PATCH 7/8] Fix reversed logic in drivers/md/md.c
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes the following compiler warning in next-20150525 using gcc 5.1.0:

drivers/md/md.c: In function ‘update_array_info’:
drivers/md/md.c:6394:26: warning: logical not is only applied to the
left hand side of comparison [-Wlogical-not-parentheses]
      !mddev->persistent  != info->not_persistent||
Signed-off-by: Eddie Kovsky <ewk@edkovsky.org>
---
 drivers/md/md.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 51f0345..4aa19e7 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -6212,7 +6212,7 @@ static int update_array_info(struct mddev *mddev, mdu_array_info_t *info)
 	    mddev->ctime         != info->ctime         ||
 	    mddev->level         != info->level         ||
 /*	    mddev->layout        != info->layout        || */
-	    !mddev->persistent	 != info->not_persistent||
+	    mddev->persistent	 == info->not_persistent||
 	    mddev->chunk_sectors != info->chunk_size >> 9 ||
 	    /* ignore bottom 8 bits of state, and allow SB_BITMAP_PRESENT to change */
 	    ((state^info->state) & 0xfffffe00)
-- 
2.6.3

